### PR TITLE
userns: use the same ID if the mapping is missing

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind bug

> /kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

#### What this PR does / why we need it:
If the mapping is missing it means it wasn't specified by the user.
In this case do not error out but use the unchanged ID.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
fix bug where runAsUser would only work with runAsGroup if userns annotations were specified
```

### CODE REVIEW

**Comment by haircommander** [comment]
seems like network flakes
/retest

**Review by giuseppe — APPROVED** [review]
/retest

File: server/container_create_linux.go [hunk 1]
```diff
@@ -214,50 +214,40 @@ func makeMountsAccessible(uid, gid int, mounts []rspec.Mount) error {
...
 		if hostID >= m.HostID && hostID < m.HostID+m.Size {
 			contID := m.ContainerID + (hostID - m.HostID)
-			return uint32(contID), nil
+			return uint32(contID)
 		}
 	}
-	return 0, fmt.Errorf("host ID %d cannot be mapped to a container ID", hostID)
+	// If the ID cannot be mapped, it means the RunAsUser or RunAsGroup was not specified
+	// so just use the original value.
+	return id
```
Comment (TomSweeneyRedHat) [inline server/container_create_linux.go:230]: Any value to a log.debug here letting you know a match wasn't found?

**Review by mrunalp — APPROVED** [review]

**Comment by umohnani8** [comment]
LGTM

**Pull request merged**